### DISCUSSION TITLE
Add ~/.kenv/bin to $PATH

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#528] [Docs]
<meta path="kit/add-kenv-to-profile">
      
# Add ~/.kenv/bin to $PATH

Each time you create a script, Script Kit also generates a command based on the name you can run from the terminal.

If you create a script named `list-downloads`, then Script Kit creates a `~/.kenv/bin/list-downloads` executable.

Then you can run the command like so in the terminal:

```bash
~/.kenv/bin/list-downloads
```

This will walk you through running the command without the full path:

```bash
list-downloads
```

**Comment by joezimjs** [comment]
is there any way to pass inputs into these commands immediately when calling them instead of waiting for the prompt? (Like with kar... but kar isn't working on my Windows machine).

**Reply by johnlindquist** [reply]
@joezimjs 

From your terminal, you should be able to invoke it like:

> This would invoke ~/.kenv/bin/foo.js with "hello-world"
```bash
foo "hello-world"
```

`kar` is based on a UNIX socket, so it only works on mac/linux (and maybe WSL?).
There's another approach that uses "run.txt" if you need a prompt:

You need to write the full path to the script, then the arguments to the run.text file, for example on mac:
```bash
echo "/Users/johnlindquist/.kenv/scripts/testing-bin-with-input.js hello" > ~/.kit/run.txt
```

This is how the Script Kit VS Code extensions works:

https://github.com/johnlindquist/kit-extension/blob/main/src/extension.ts#L99

**Reply by joezimjs** [reply]
If I run  `switch-audio-output` in powershell or cmd, it works fine. If I run `switch-audio-output "a"` in powershell or cmd I get the following error:

```
Error: Command failed with exit code 1: a
'a' is not recognized as an internal or external command,
operable program or batch file.
    at makeError (file:///C:/Users/JoeZimmerman/.kit/node_modules/@johnlindquist/globals/dist/index.js:36:111621)
    at file:///C:/Users/JoeZimmerman/.kit/node_modules/@johnlindquist/globals/dist/index.js:36:124733
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///C:/Users/JoeZimmerman/.kenv/scripts/switch-audio-output.js?uuid=8157203d-28e3-4425-aaaa-79955511ca94:25:1 {
  shortMessage: 'Command failed with exit code 1: a',
  command: 'a',
  escapedCommand: 'a',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: "'a' is not recognized as an internal or external command,\r\n" +
    'operable program or batch file.',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
```

That second option of using run.txt appears to be doing nothing at all, and of course because of how that mechanism works, there's absolutely no transparency into why it's not working.

**Comment by ramiroaraujo** [comment]
Update for this old post:
if you have `kenvs` you then (also) have to add their `bin` path to `PATH`

**Reply by joezimjs** [reply]
Not sure what "have `kenvs`" means...
